### DISCUSSION TITLE
How do I call the function?

### DISCUSSION THREAD

**Discussion by devilish-crow** [cc-tweaked/CC-Tweaked#1412] [General]
Sorry because my main programming language is not lua, but how do I call the function in cc:tweaked ?

Here is my code :

function CUI(m)
local n=1
local l=#m
while true do
term.clear()
term.setCursorPos(1,2)
for i=1, l, 1 do
if i==n then print(i, "["..m[i].."]") else print(i, m[i]) end
end
print("Select a number[arrow up/arrow down]")
local a, b= os.pullEventRaw()
if a == "key" then
if b==200 and n>1 then n=n-1 end
if b==208 and n<=l then n=n+1 end
if b==28 then break end
end
end
term.clear() term.setCursorPos(1,1)
return n
end
function CUI(m)

**Comment by Lupus590** [comment]
Mostly like that, but `m` should be a table (it's currently null).

You create tables like this.
```lua
local myTable = {"hello", "world"}
```
There's loads more that tables can do but that's the most basic way that matches how your function seems to be using it.

> my main programming language is not lua

What is your main programming language? I might be able to translate some of its concepts for you and help you with learning Lua.

**Reply by devilish-crow** [reply]
My main languages are JavaScript and C.

**Reply by devilish-crow** [reply]
Can I install lua modules on cc:tweaked?

**Reply by fatboychummy** [reply]
> Can I install lua modules on cc:tweaked?

As in Lua C modules? No, CC is written in Java.
